infrastructure/lib/lambda-stack.ts

```diff
@@ -0,0 +1,40 @@
+import * as ecs_patterns from '@aws-cdk/aws-ecs-patterns';
+import * as cdk from '@aws-cdk/core';
+import { Code, Function, Runtime } from '@aws-cdk/aws-lambda';
+import { Table } from '@aws-cdk/aws-dynamodb';
+import * as assets from '@aws-cdk/aws-s3-assets';
+
+export interface LambdaStackProps extends cdk.StackProps {
+    readonly table: Table;
+}
+
+export class LambdaStack extends cdk.Stack {
+  readonly lambda: Function;
+
+  constructor(scope: cdk.Construct, id: string, props: LambdaStackProps) {
+        super(scope, id, props);
+
+        const zipFileName = 'cribbage-lambda';
+
+        const asset = new assets.Asset(this, 'CribbageZip', {
+            path: '../' + zipFileName + '.zip',
+        });
```

Comment:
Does this upload the zip to an S3 bucket? If so, I'd love to name it after the commit hash instead so we can easily deploy a specific version of the app. Maybe overkill for a hobby project, but would be nice to have.

Reply:
And if not, we should upload the zip to an S3 bucket

Reply:
> An asset represents a local file or directory, which is automatically uploaded to S3 and then can be referenced within a CDK application.

Reply:
Cool. Can we change the name from `CribblyZip` to the git hash?

Reply:
@joshprzybyszewski ^?